packages/repack/src/modules/ScriptManager/__tests__/ScriptManager.test.ts

```diff
@@ -832,6 +839,43 @@ describe('ScriptManagerAPI', () => {
     spy.mockReset();
     spy.mockRestore();
   });
+
+  it('should refetch failed script', async () => {
+    jest.useFakeTimers({ advanceTimers: true });
+    const spy = jest.spyOn(NativeScriptManager, 'loadScript');
+
+    spy.mockImplementation(
+      (_scriptId: string, _scriptConfig: NormalizedScriptLocator) =>
+        Promise.reject({ code: 'NetworkFailed' })
+    );
+
+    const cache = new FakeCache();
+    ScriptManager.shared.setStorage(cache);
+
+    ScriptManager.shared.addResolver(async (scriptId, _caller) => {
+      return {
+        url: Script.getRemoteURL(scriptId),
+        cache: true,
+      };
+    });
+
+    await expect(async () =>
+      ScriptManager.shared.loadScript('miniApp')
+    ).rejects.toThrow();
+
+    // //expected to cache be empty
+    expect(Object.keys(cache.data).length).toBe(0);
+
+    await expect(async () =>
+      ScriptManager.shared.loadScript('miniApp')
+    ).rejects.toThrow();
+
+    // expected to fetch again
+    // expect(spy.mock.lastCall?.[1].fetch).toBe(true);
+
+    spy.mockReset();
+    spy.mockRestore();
```

Comment:
`mockRestore()` should be sufficient on it's own

https://jestjs.io/docs/mock-function-api#mockfnmockrestore